spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) 2020 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phase, such as dev, testing, live, etc. 
+
+=== Specify Config Profile
+
+The config profile can be specified via a system property `mp.config.profile`. It can be set when starting your microservice. The value of the property can contain only characters that are valid for config property names. 
+This is because the name of the profile is directly stored in the name of the config property. e.g.
+
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+The property `mp.config.profile` is only read once on microservice starting. Any changes after that is ignored. Furthermore, the value specifies a single profile. Only one profile can be active at a time.
+Commas in the value have no special meaning. For example, if the value of `mp.config.profile` is `testing,live`,  a single profile named `testing,live` is enabled instead of enabling two profiles. 
+
+=== How Config Profile works
+
+In supporting the Config Profile, the config property names will need to start with the naming convention of `%configProfile.`, (e.g. `%live.vehicle.name`).
```

Comment:
Do you realize that `%` is not allowed in some config sources, e.g. in environment variable names? For environment variables, the "profile" property would need to be specified using `_` instead of `%`, such as: `_live_vehicle_name`. I'm not sure if it would cause any problems but we should think about it.

Reply:
It hasn't been a problem for us; generally users have not seemed to have needed to use env vars (or system properties for that matter) to override profile-specific properties.

Reply:
However if it is to be usable in all Config Sources then perhaps we should think of a different identifier.

Reply:
If it is specified as environment variables, it will be transformed from % to _. It is same as other config properties.

Reply:
Since environment variables would map `%` to  `_`, can't we just use `_` as the prefix also for system properties instead of introducing a new special character `%`? Implementations can support alternative notations like using `%`, no big deal. But a standard notation `_configProfile.property` would be more consistent with the current MP Config specification and wouldn't introduce a new special character.

Reply:
For the record, the profile property names were discussed here: https://github.com/eclipse/microprofile-config/pull/554#discussion_r409668376 and this thread continues that discussion.